share/src/main/java/org/apache/kafka/server/share/persister/Persister.java

```diff
@@ -40,15 +40,15 @@ public interface Persister {
      * @param request Request parameters
      * @return A {@link CompletableFuture} that completes with the result.
      */
-    CompletableFuture<ReadShareGroupStateResult> readState(ReadShareGroupStateParameters request) throws IllegalArgumentException;
+    CompletableFuture<ReadShareGroupStateResult> readState(ReadShareGroupStateParameters request);
 
     /**
      * Write share-partition state.
      *
      * @param request Request parameters
      * @return A {@link CompletableFuture} that completes with the result.
      */
-    CompletableFuture<WriteShareGroupStateResult> writeState(WriteShareGroupStateParameters request) throws IllegalArgumentException;
```

Comment:
thanks @chia7712, created issue https://issues.apache.org/jira/browse/KAFKA-18148, I think we can target it for 4.1 since these APIs are not implemented yet in `DefaultStatePersister`